src/features/chart/components/Chart.tsx

```diff
@@ -53,25 +55,28 @@ const formatDate = (isoDate, ChartRange) => {
 export const Chart = () => {
   const dispatch = useDispatch();
   const { selectedChartData, selectedChartRange } = useSelector(
-    state => state.chartData
+    (state: GlobalState) => state.chartData
   );
-  const { selectedStockTicker } = useSelector(state => state.stockTickerData);
-  const updateChartRange = stock => dispatch(updateChartAction(stock));
-  const onClickHandler = event => {
-    updateChartRange(event.target.value);
+  const { selectedStockTicker } = useSelector(
+    (state: GlobalState) => state.stockTickerData
+  );
+  const updateChartRange = (stock: string) =>
+    dispatch(updateChartRangeAction(stock));
+  const onClickHandler: React.MouseEventHandler<HTMLButtonElement> = event => {
+    updateChartRange(event.currentTarget.value);
   };
 
-  const chartData =
-    selectedChartData &&
-    selectedChartData.map(data => ({
-      date: formatDate(data.date, selectedChartRange),
-      close: data.close
-    }));
+  const chartData = selectedChartData
```

Comment:
FYI: This is perfectly okay, but there's another (potentially prettier) way of writing it, too:
```ts
const chartData = (selectedChartData || []).map(data => ({
  ...
```
It's a matter of preference, of course.